### PR TITLE
feat(router): migrate `RouterLinkWithHref` references to `RouterLink`

### PR DESCRIPTION
**Pull request by AndrewKushnir**
Since Angular v15, the `RouterLink` contains the logic of the `RouterLinkWithHref` directive and now developers can always import and use the `RouterLink` directive when they need to add a `[routerLink]` in templates. This migration finds all imports and usages of the `RouterLinkWithHref` class and rewrites them to `RouterLink` instead.

```ts
import { RouterLinkWithHref } from '@angular/router';

@Component({
  standalone: true,
  template: `<a [routerLink]="'/abc'">`,
  imports: [RouterLinkWithHref]
})
export class MyComponent {
  @ViewChild(RouterLinkWithHref) aLink!: RouterLinkWithHref;
}
```

```ts
import { RouterLink } from '@angular/router';

@Component({
  standalone: true,
  template: `<a [routerLink]="'/abc'">`,
  imports: [RouterLink]
})
export class MyComponent {
  @ViewChild(RouterLink) aLink!: RouterLink;
}
```

## PR Type
What kind of change does this PR introduce?

- [x] Feature

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by atscott — APPROVED** [review]

File: packages/core/schematics/migrations/router-link-with-href/util.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+          printer.printNode(ts.EmitHint.Unspecified, rewrittenNamedImports, sourceFile);
+      rewrite(
+          routerLinkNamedImports.getStart(), routerLinkNamedImports.getWidth(), replacementText);
+    }
+  } else {
+    // The `RouterLink` symbol is not imported, rewrite `RouterLinkWithHref` -> `RouterLink`.
+    rewrite(routerLinkWithHrefSpec.getStart(), routerLinkWithHrefSpec.getWidth(), routerLink);
+  }
+
+  for (const usage of usages) {
```
Comment (crisbeto) [inline packages/core/schematics/migrations/router-link-with-href/util.ts:66]: Should we run these replacements in reverse? I think that the node offsets are going to shift around further down in the file.

**Review by crisbeto — APPROVED** [review]
LGTM

**Comment by AndrewKushnir** [comment]
@crisbeto thanks for the review 👍

I've actually rewrote the logic a bit taking into account your comment. Now the process looks like this:
- we find all usages and imports, store them in an array (as start pos, width of the symbol and a replacement)
- sort by start pos in descending order (last-to-first usages) to avoid offset shifts
- once all usages are collected and sorted, we apply rewrites

Previously, lookups and rewrites were interleaved.

Could you please take another look when you get a chance?

**Review by crisbeto — APPROVED** [review]
LGTM

**Comment by AndrewKushnir** [comment]
[Presubmit](http://test/OCL:479347096:BASE:479347170:1665076943668:6a11c4b7).

**Comment by jessicajaniuk** [comment]
This PR was merged into the repository by commit 16c8f55663c30270fcd647b1a8a20ddbc8923349.

**Pull request closed without merging**